File: Internal/Content/Account/SelfUserContent.cs
using BonfireServer.Internal.Common;
using BonfireServer.Internal.Content.Server;

namespace BonfireServer.Internal.Content.Account;

public class SelfUserContent(User user) : IBaseContent
{
    public User User { get; set; } = user;
    public string Email { get; set; } = user.Email;
    public Dictionary<ServerInfoContent, string> Nicknames { get; set; } = user.Nicknames.ToDictionary(x => new ServerInfoContent(x.Key), x => x.Value);
    public List<ServerInfoContent> Servers { get; set; } = user.Servers.Select(x => new ServerInfoContent(x)).ToList();
    public List<User> Friends { get; set; } = user.Friends;
    public List<User> FriendRequests { get; set; } = user.FriendRequests;
    public List<ChannelInfoContent> DirectMessages { get; set; } = user.DirectMessages.Select(x => new ChannelInfoContent(x)).ToList();
    public List<ChannelInfoContent> Channels { get; set; } = user.Servers
        .SelectMany(server => server.Channels.Select(channel => new ChannelInfoContent(channel)))
        .ToList();
}

File: Public/Scripts/Content/Account/SelfUserContent.ts
import {User} from "../../Common/User.js";
import {Server} from "../../Common/Server.js";
import {Channel} from "../../Common/Channel.js";

export class SelfUserContent
{
    public User: User;
    public Email: string;
    public Nicknames: Map<Server, string>;
    public Servers: Server[];
    public Channels: Channel[];
    public Friends: User[];
    public FriendRequests: User[];
    public DirectMessages: Channel[];

    constructor(
        user: User,
        email: string,
        nicknames: Map<Server, string>,
        servers: Server[],
        channels: Channel[],
        friends: User[],
        friendRequests: User[],
        directMessages: Channel[],
    ) {
        this.User = user;
        this.Email = email;
        this.Nicknames = nicknames;
        this.Servers = servers;
        this.Channels = channels;
        this.Friends = friends;
        this.FriendRequests = friendRequests;
        this.DirectMessages = directMessages;
    }
}